### DISCUSSION TITLE
[Question] 如何播放二进制流

### DISCUSSION THREAD

**Discussion by cenumi** [wang-bin/fvp#149] [General]
我们的直播视频流是从 websocket 中获取的, 并且需要发送消息, 现在临时的解决方法是收到流后写入文件, 传入文件地址后以 'avio.follow', '1' 的方式进行播放, 效率会有些影响. 请问有什么解决方案吗.

在目前的issue中搜到了 
mem: 方式但似乎只能传入整个文件
fd: 似乎在 flutter 中无法使用
enqueue 方法在 fvp 中不存在

十分感谢

**Comment by wang-bin** [comment]
`fd:`使用上有什么问题

**Comment by cenumi** [comment]
> `fd:`使用上有什么问题

在 Flutter 中似乎没有对应的API，也可能是我没找到相关的用法，求教了

**Comment by wang-bin** [comment]
估计得写c代码吧

**Comment by Nicoeevee** [comment]
> > `fd:`使用上有什么问题
> 
> 在 Flutter 中似乎没有对应的API，也可能是我没找到相关的用法，求教了

https://github.com/arthenica/ffmpeg-kit/issues/808#issuecomment-1715710379

**Comment by cenumi** [comment]
> 
> [arthenica/ffmpeg-kit#808 (comment)](https://github.com/arthenica/ffmpeg-kit/issues/808#issuecomment-1715710379)

本地tcp转发确实是一种方案，但终究是折中的方案，也增加了成本，我们这边一屏上可能有十几个播放器，暂时这边先用rtmp流了，这种功能可能还是得原生实现，这边在做的平台是windows，不是很熟悉windows原生，所以来请教下的

**Comment by cenumi** [comment]
mdk 中已经实现 appendBuffer 方法，现在可以暴露给 fvp 使用吗 :heart_eyes:
